perses/samplers/samplers.py

```diff
@@ -1174,8 +442,10 @@ def update_state(self):
 
         # Propose new chemical state.
         if self.verbose: print("Proposing new topology...")
-        [system, topology, positions] = [self.sampler.thermodynamic_state.system, self.topology, self.sampler.sampler_state.positions]
-        topology_proposal = self.proposal_engine.propose(system, topology)
+        [system, topology, positions] = [self.sampler.thermodynamic_state.get_system(remove_thermostat=True), self.topology, self.sampler.sampler_state.positions]
```

Comment:
This seems like something we might want to add to [`ThermodynamicState`](https://openmmtools.readthedocs.io/en/latest/api/generated/openmmtools.states.ThermodynamicState.htm) as a `get_openmm_topology()` method that takes an optional `SamplerState` argument to grab the box vectors.